server/filestore.go

```diff
@@ -8679,6 +8749,18 @@ func (fs *fileStore) Truncate(seq uint64) error {
 		fs.mu.Unlock()
 		return err
 	}
+
+retry:
+	nlmbcache := nlmb.cache.Value()
```

Comment:
`nlmb.mu.Lock()` is not held here.

Actually we should already be holding it when we do `nlmb.enableForWriting(fs.fip)` above.. (I know that was not touched in this PR).
Could you lock before `enableForWriting` and unlock after we exit below if branch?